hibernate-core/src/main/java/org/hibernate/sql/ast/tree/from/UnionTableReference.java

```diff
@@ -67,9 +67,6 @@ public TableReference getTableReference(
 	}
 
 	private boolean hasTableExpression(String tableExpression) {
-		if ( tableExpression.equals( getTableExpression() ) ) {
-			return true;
-		}
```

Comment:
Since `subclassTableSpaceExpressions` already contains the `tableExpression` this should be fine.